internal/provider/dns_settings_data_source.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	netbird "github.com/netbirdio/netbird/management/client/rest"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &DNSSettingsDataSource{}
+
+func NewDNSSettingsDataSource() datasource.DataSource {
+	return &DNSSettingsDataSource{}
+}
+
+// DNSSettingsDataSource defines the data source implementation.
+type DNSSettingsDataSource struct {
+	client *netbird.Client
+}
+
+func (d *DNSSettingsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_settings"
+}
+
+func (d *DNSSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Read DNS Management Settings",
+		Attributes: map[string]schema.Attribute{
+			"extra_dns_labels": schema.ListAttribute{
```

Comment:
shouldn't it be `disabled_management_groups` ?

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7